shap/explainers/_tree.py

```diff
@@ -182,7 +182,7 @@ def __init__(
             self.data = data
         if self.data is None:
             feature_perturbation = "tree_path_dependent"
-            #warnings.warn("Setting feature_perturbation = \"tree_path_dependent\" because no background data was given.")
+            warnings.warn("Setting feature_perturbation = \"tree_path_dependent\" because no background data was given.")
```

Comment:
This change may impact a large fraction of users, so I think we should carefully consider what level of notification is appropriate. Would `logging.info` be more appropriate here?

In the logging [HOWTO guide](https://docs.python.org/3/howto/logging.html), they recommend:

- To report events that occur "during normal operation of a program", use logging `info()`
- To issue a warning regarding a particular runtime event, if the issue is avoidable and the client application should be modified to eliminate the warning, use `warnings.warn`

To me, this seems like normal operation rather than an "issue", as this behaviour is described in the docs of tree explainer:

```
feature_perturbation : "interventional" (default) or "tree_path_dependent" (default when data=None)
```

My guess is that Scott commented this warning as it was becoming a bit annoying.